df.art.xml

```diff
@@ -587,61 +587,92 @@
     </struct-type>
 
-    <struct-type type-name='musical_form_vocals'>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <stl-vector type-name='int32_t'/>
-        <stl-vector name='vocal_components' type-name='int32_t'/> 0: melody, 1: counterpoint, 2: harmony, 3: rhythm
-        <stl-vector name='phrase_lengths' type-name='int32_t'/> 0: short, 1: mid-length, 2: long, 3: varied length
-        <stl-vector type-name='int32_t'/>
-        <stl-vector type-name='int32_t'/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <int32_t/>
-        <stl-vector pointer-type='musical_form_melodies'/>
-        <int32_t/>
-        <int32_t/>
+    <struct-type type-name='musical_form_melodies'>
+        <enum name='style' type-name='musical_form_melody_style' base-type='int32_t'/>
+        <enum name='frequency' type-name='musical_form_melody_frequency' base-type='int32_t'/>
+        <stl-vector name='intervals' pointer-type='musical_form_interval'/>
+        <bitfield name='features' type-name='musical_form_feature'/>
+    </struct-type>
+
+    <struct-type type-name='musical_form_passage'>
+        <enum name='type' type-name='musical_form_passage_type' base-type='int32_t'/>
+        <int32_t name='passage_reference' init-value='-1' comment="used when doing Exposition, Recapitualation, Synthesis, and Variation"/>
+        <int32_t name='passage_range_end' init-value='-1' comment="when doing Synthesis of a range of passages"/>
+        <int32_t name='unk_4' comment="'min_times' for a 3-5 range, but doesn't match up with 1 for both repeat 2 times and no repeat mentioned"/>
+        <int32_t name='unk_5' comment="'max_times' for a 3-5 range, but doesn't match up with 1 for both repeat 2 times and no repeat mentioned"/>
+        <int32_t name='poetic_form_id' ref-target='poetic_form'/>
+        <int32_t name='written_content_id' ref-target='written_content' comment="suspect bug in exported legends (and possibly DF itself) as no mentioning of the poems (or any alternative) referenced here were mentioned in the two entries examined"/>
+        <int32_t name='scale_id' ref-target='scale'/>
+        <int32_t name='scale_sub_id' comment="references the scales element of the scale"/>
+        <int32_t name='rhythm_id' ref-target='rhythm'/>
+        <int32_t name='sub_rhythm' comment="Guess, based on the pattern above"/>
+        <int32_t name='rhythm_pattern' comment="references the patterns element of rhythm"/>
+        <stl-vector name='instruments' type-name='int32_t' comment="indices into the instruments vector"/>
+        <stl-vector name='components' type-name='musical_form_passage_component_type' base-type='int32_t'/>
+        <stl-vector name='passage_lengths' type-name='musical_form_passage_length_type' base-type='int32_t'/>
```

Comment:
Unfortunately, for these, if you want to specify a `base-type`, you need a `<enum>` as a child of the `<stl-vector>` instead.